plugins/modules/ndo_fabric_span_session.py

```diff
@@ -0,0 +1,698 @@
+    mtu = module.params.get("mtu")
+    destination_epg = module.params.get("destination_epg")
+    destination_port = module.params.get("destination_port")
+    destination_port_channel = module.params.get("destination_port_channel")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "monitoring_tenant", template_name, template_id)
+    mso_template.validate_template("monitoring")
+    object_description = "SPAN Session "
+    site_id = mso_template.template.get("monitoringTemplate").get("sites")[0].get("siteId")
+
+    match = None
+    if uuid or name:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            mso_template.template.get("monitoringTemplate", {}).get("template", {}).get("spanSessions", []),
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+    else:
+        match = mso_template.template.get("monitoringTemplate", {}).get("template", {}).get("spanSessions", [])
+
+    if (uuid or name) and match:
+        set_fabric_span_session_object_details(mso_template, site_id, match.details)
+        mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = [set_fabric_span_session_object_details(mso_template, site_id, obj) for obj in match]  # Query all objects
+
+    if state != "query":
+        span_session_path = "/monitoringTemplate/template/spanSessions/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        mso_values = dict()
+        if destination_epg:
+            mso_values["destination"] = dict(
+                remote=dict(
+                    epgRef=get_epg_uuid(mso_schemas, destination_epg.get("epg"), destination_epg.get("epg_uuid")),
+                    spanVersion=destination_epg.get("span_version"),
+                    enforceSpanVersion=destination_epg.get("enforce_span_version"),
+                    destIPAddress=destination_epg.get("destination_ip"),
+                    srcIPPrefix=destination_epg.get("source_ip_prefix"),
+                    flowID=destination_epg.get("flow_id"),
+                    ttl=destination_epg.get("ttl"),
+                    dscp=TARGET_DSCP_MAP.get(destination_epg.get("dscp")),
+                ),
+                mtu=mtu,
+            )
+
+        if destination_port:
+            # Destination Port supports UUIDs, but we have no module to get them
+            destination_port_uuid = destination_port.get("port_uuid")
+            if destination_port_uuid is None:
+                node = destination_port.get("port").get("node")
+                interface_port = destination_port.get("port").get("path")
+
+                site_data = mso.request(
```

Comment:
Can the util function `get_site_interface_detail` you create be used here?

Reply:
Done!